solutions/silver/cf-1732C1.mdx

```diff
@@ -0,0 +1,159 @@
+---
+id: cf-1732C1
+source: CF
+title: "Sheikh (Easy version)"
+author: Ryan Chou
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/108327)
+
+<Spoiler title="Hint 1">
+What's the maximum value of $f(l, r) = \operatorname{sum}(l, r) - \operatorname{xor}(l, r)$?
+</Spoiler>
+
+<Spoiler title="Hint 1 (answer)">
+Note that the XOR of two numbers can never grow _more_ than their sum.
```

Comment:
```suggestion
Note that the XOR of two numbers can never grow _more_ than their sum. Mathematically, this can be expressed as:

$a-b \le (a+c)-(b \bigoplus c)$
```

Reply:
waitt what is c

Reply:
okay so your current explanation isn't making sense

> suppose our current sum is $a$ and xor is $b$. Then if we consider including $c$ to the range, we know that the difference can only increase.

This is the key observation

Reply:
the first step in figuring this out is that $a \oplus b \leq a + b$, and that property is implied.

it's confusing rn because ur adding it right after the "xor of a and b is leq than their sum", so ur acting like this inequality is equivalent to that.

do you think the transition from "xor of two numbers leq their sum" -> "adding more numbers can only increase the answer"?

Reply:
not quite surew what the question

Reply:
oops edited

Reply:
Yes it is too confusing.

Reply:
just made a commit -- does it make sense? or do you think adding the math thingy too would be better?

i js don't wanna confuse ppl :>